pkg/build/builder/docker.go

```diff
@@ -64,16 +60,10 @@ func (d *DockerBuilder) Build() error {
 		return err
 	}
 	glog.V(4).Infof("Starting Docker build from build config %s ...", d.build.Name)
-	var push bool
-
 	// if there is no output target, set one up so the docker build logic
-	// will still work, but we won't push it at the end.
+	// (which requires a tag) will still work, but we won't push it at the end.
 	if d.build.Spec.Output.To == nil || len(d.build.Spec.Output.To.Name) == 0 {
-		d.build.Spec.Output.To = &kapi.ObjectReference{
-			Kind: "DockerImage",
-			Name: noOutputDefaultTag,
-		}
-		push = false
+		d.build.Status.OutputDockerImageReference = d.build.Name
```

Comment:
We set B instead of A because we used to consume A and now we consume B.

And we never made a promise about what was in the To field.  We set it for our own tagging purposes.  We changed the code that consumes the value, so we changed what we set.